core/model/modx/mysql/modcontext.class.php

```diff
@@ -12,33 +12,88 @@ class modContext_mysql extends modContext {
     public static function getResourceCacheMapStmt(&$context) {
         $stmt = false;
         if ($context instanceof modContext) {
+            $time=microtime(true);
+            $cache_alias_map = $context->getOption('cache_alias_map');
+            $use_context_resource_table = $context->getOption('use_context_resource_table',null,1);
+
             $tblResource= $context->xpdo->getTableName('modResource');
             $tblContextResource= $context->xpdo->getTableName('modContextResource');
+
             $resourceFields= array('id','parent','uri');
+
+            // we do not need to select uri if cache_alias_map is set to false
+            if ($cache_alias_map==0) $resourceFields = array('id','parent');
+
             $resourceCols= $context->xpdo->getSelectColumns('modResource', 'r', '', $resourceFields);
-            $bindings = array($context->get('key'), $context->get('key'));
-            $sql = "SELECT {$resourceCols} FROM {$tblResource} `r` FORCE INDEX (`cache_refresh_idx`) LEFT JOIN {$tblContextResource} `cr` ON `cr`.`context_key` = ? AND `r`.`id` = `cr`.`resource` WHERE `r`.`id` != `r`.`parent` AND (`r`.`context_key` = ? OR `cr`.`context_key` IS NOT NULL) AND `r`.`deleted` = 0 GROUP BY `r`.`parent`, `r`.`menuindex`, `r`.`id`, `r`.`uri`";
+            $contextKey = $context->get('key');
+
+            $bindings = array();
+            $sql  = "SELECT {$resourceCols} FROM {$tblResource} `r` ";
+            if ($use_context_resource_table) {
+                $bindings = array($contextKey, $contextKey);
+                $sql .= "FORCE INDEX (`cache_refresh_idx`) ";
+                $sql .= "LEFT JOIN {$tblContextResource} `cr` ON `cr`.`context_key` = ? AND `r`.`id` = `cr`.`resource` ";
+            }
+            $sql .= "WHERE `r`.`deleted` = 0 "; //"AND `r`.`id` != `r`.`parent`";
+            if ($use_context_resource_table) {
+                $sql .= "AND (`r`.`context_key` = ? OR `cr`.`context_key` IS NOT NULL) ";
+                $sql .= "GROUP BY `r`.`parent`, `r`.`menuindex`, `r`.`id`, `r`.`uri` ";
+            } else {
+                $bindings = array($contextKey);
+                $sql .= "   AND `r`.`context_key` = ?";
+            }
+
             $criteria = new xPDOCriteria($context->xpdo, $sql, $bindings, false);
             if ($criteria && $criteria->stmt && $criteria->stmt->execute()) {
                 $stmt =& $criteria->stmt;
             }
+
+            // output warning if query is too slow
+            $time = ((microtime(true)-$time));
+            if ($time >= 1.0 && $use_context_resource_table==1) $context->xpdo->log(2,"[modContext_mysql] Slow query detected. Consider to set 'use_context_resource_table' to false.");
```

Comment:
Code style here as well.

Instead of log level 2, please use the xPDO level constants so it's clear from the code what level it is logging at.